File: cakk-api/src/main/java/com/cakk/api/mapper/BusinessInformationMapper.java
package com.cakk.api.mapper;

import java.util.List;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import com.cakk.api.dto.param.operation.OwnerCandidateParam;
import com.cakk.api.dto.response.shop.CakeShopOwnerCandidateResponse;
import com.cakk.api.dto.response.shop.CakeShopOwnerCandidatesResponse;
import com.cakk.domain.mysql.entity.user.BusinessInformation;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class BusinessInformationMapper {

	public static CakeShopOwnerCandidatesResponse supplyCakeShopOwnerCandidatesResponseBy(
		List<BusinessInformation> businessInformations
	) {
		List<OwnerCandidateParam> candidates = businessInformations
			.stream()
			.map(businessInformation ->
				OwnerCandidateParam.builder()
					.userId(businessInformation.getUser().getId())
					.nickname(businessInformation.getUser().getNickname())
					.profileImageUrl(businessInformation.getUser().getProfileImageUrl())
					.email(businessInformation.getUser().getEmail())
					.timestamp(businessInformation.getUpdatedAt())
					.build())
			.toList();

		return new CakeShopOwnerCandidatesResponse(candidates);
	}

	public static CakeShopOwnerCandidateResponse supplyCakeShopOwnerCandidateResponseBy(
		BusinessInformation businessInformation
	) {
		return CakeShopOwnerCandidateResponse.builder()
			.userId(businessInformation.getUser().getId())
			.cakeShopId(businessInformation.getCakeShop().getId())
			.email(businessInformation.getUser().getEmail())
			.businessRegistrationImageUrl(businessInformation.getBusinessRegistrationImageUrl())
			.idCardImageUrl(businessInformation.getIdCardImageUrl())
			.emergencyContact(businessInformation.getEmergencyContact())
			.build();
	}
}

File: cakk-admin/src/main/kotlin/com/cakk/admin/mapper/BusinessInformationMapper.kt
package com.cakk.admin.mapper

import com.cakk.admin.dto.param.OwnerCandidateParam
import com.cakk.admin.dto.response.CakeShopOwnerCandidateResponse
import com.cakk.admin.dto.response.CakeShopOwnerCandidatesResponse
import com.cakk.domain.mysql.entity.user.BusinessInformation

fun supplyCakeShopOwnerCandidatesResponseBy(businessInformationList: List<BusinessInformation>): CakeShopOwnerCandidatesResponse {
    val candidates: List<OwnerCandidateParam> = businessInformationList
        .map { businessInformation: BusinessInformation ->
            OwnerCandidateParam(
                userId = businessInformation.user.id,
                nickname = businessInformation.user.nickname,
                profileImageUrl = businessInformation.user.profileImageUrl,
                email = businessInformation.user.email,
                timestamp = businessInformation.updatedAt
            )
        }.toList()

    return CakeShopOwnerCandidatesResponse(candidates)
}

fun supplyCakeShopOwnerCandidateResponseBy(businessInformation: BusinessInformation): CakeShopOwnerCandidateResponse {
    return CakeShopOwnerCandidateResponse(
        userId = businessInformation.user.id,
        cakeShopId = businessInformation.cakeShop.id,
        email = businessInformation.user.email,
        businessRegistrationImageUrl = businessInformation.businessRegistrationImageUrl,
        idCardImageUrl = businessInformation.idCardImageUrl,
        emergencyContact = businessInformation.emergencyContact
    )
}
